src/det.jl

```diff
@@ -21,6 +21,23 @@ end
     return vecdot(x0, cross(x1, x2))
 end
 
+@inline function _det(::Size{(4,4)}, A::StaticMatrix,S::Type)
+    A = similar_type(A,S)(A)
```

Comment:
There used to be compiler problems with variable names that change types. Has this been fixed? (I.e. is code optimal here when `eltype(A) != S`?)